paasta_tools/paasta_serviceinit.py

```diff
@@ -64,39 +72,52 @@ def main():
     else:
         log.setLevel(logging.WARNING)
 
+    instances = []
     command = args.command
-    service_instance = args.service_instance
-    service, instance, _, __ = decompose_job_id(service_instance)
-
-    cluster = load_system_paasta_config().get_cluster()
-    instance_type = validate_service_instance(service, instance, cluster, args.soa_dir)
-    if instance_type == 'marathon':
-        return_code = marathon_serviceinit.perform_command(
-            command=command,
-            service=service,
-            instance=instance,
-            cluster=cluster,
-            verbose=args.verbose,
-            soa_dir=args.soa_dir,
-            app_id=args.app_id,
-            delta=args.delta,
-        )
-        sys.exit(return_code)
-    elif instance_type == 'chronos':
-        return_code = chronos_serviceinit.perform_command(
-            command=command,
-            service=service,
-            instance=instance,
-            cluster=cluster,
-            verbose=args.verbose,
-            soa_dir=args.soa_dir,
-        )
-        sys.exit(return_code)
+    if (args.service_instance):
+        service_instance = args.service_instance
+        service, instance, _, __ = decompose_job_id(service_instance)
+        instances.append(instance)
+    elif (args.service and args.instances):
+        service = args.service
+        instances = args.instances.split(',')
     else:
-        log.error("I calculated an instance_type of %s for %s which I don't know how to handle. Exiting."
-                  % (instance_type, compose_job_id(service, instance)))
+        log.error("The name of service or the name of instance to inspect is missing. Exiting.")
         sys.exit(1)
 
+    cluster = load_system_paasta_config().get_cluster()
+    for instance in instances:
+        print("instance:%s" % instance)
+        instance_type = validate_service_instance(service, instance, cluster, args.soa_dir)
+        if instance_type == 'marathon':
+            return_code = marathon_serviceinit.perform_command(
+                command=command,
+                service=service,
+                instance=instance,
+                cluster=cluster,
+                verbose=args.verbose,
+                soa_dir=args.soa_dir,
+                app_id=args.app_id,
+                delta=args.delta,
+            )
+        elif instance_type == 'chronos':
+            return_code = chronos_serviceinit.perform_command(
+                command=command,
+                service=service,
+                instance=instance,
+                cluster=cluster,
+                verbose=args.verbose,
+                soa_dir=args.soa_dir,
+            )
+        else:
+            log.error("I calculated an instance_type of %s for %s which I don't know how to handle. Exiting."
+                      % (instance_type, compose_job_id(service, instance)))
+            return_code = 1
+
+        if return_code:
```

Comment:
This is a bit odd because we are just overwriting `return_code` in the loop.

Can you return the max() of the return codes? I'm pretty sure there are other places in our code where we do this...